examples/Huggingface_Transformers/Image_classification_docker/HF-only/README.md

```diff
@@ -0,0 +1,158 @@
+
+
+# HF only
+
+## Setup
+Log into the terminal of an AWS EC2 `t2.micro` instance with a base AWS Deep Learning PyTorch AMI (as it includes [`torch-model-archiver`](https://github.com/pytorch/serve/tree/master/model-archiver) and `docker`)
+
+The `Using Torchserve Docker containers on EC2 for inference` section of [this Notion blog](https://www.notion.so/Day-2-3-Torchserve-custom-handlers-and-Docker-containers-02665de910a64aedab2b907a9a0cc9b0#3825849dfb8942379df2cdce8a729d9a) by me can be followed for logging into the appropriate AWS EC2 instance.
+
+
+## Running the Torchserve server
+Within the AWS EC2 instance, activate the PyTorch environment and navigate to the current folder after cloning this repo: 
+
+```
+source activate pytorch
+git clone https://github.com/tripathiarpan20/HF-torchserve-pipeline
+cd HF-torchserve-pipeline/HF-only/
+```
+
+Build a Torchserve CPU Docker container (or GPU/IPEX containers following the [original guide](https://github.com/pytorch/serve/tree/master/docker#create-torchserve-docker-image)):
+```
+git clone https://github.com/pytorch/serve.git
+cd serve/docker
+./build_image.sh -bt production -t torchserve-cpu-prod
+cd ../..
+rm -rf serve
+```
+
+Check whether the Torchserve image is present in the list of Docker images:
+```
+docker images
+```
+
+Run the Torchserve server container with Docker and archived model (refer to [this](https://github.com/pytorch/serve/tree/master/docker#create-torch-model-archiver-from-container) and [this](https://github.com/pytorch/serve/blob/fd4e3e8b72bed67c1e83141265157eed975fec95/docs/use_cases.md#secure-model-serving) for more details):
+
+```
+mkdir -p HF-models
+mkdir -p model-store
+docker run -d --rm -it --shm-size=50g -p 8080:8080 -p 8081:8081 --name torchserve-cpu-prod --mount type=bind,source=$(pwd)/scripts/config.properties,target=/home/model-server/config.properties --mount type=bind,source=$(pwd)/model-store,target=/home/model-server/model-store --mount type=bind,source=$(pwd)/HF-models,target=/home/model-server/HF-models torchserve-cpu-prod torchserve --ncs --model-store=/home/model-server/model-store --ts-config config.properties
+```
+
+Check whether the model was started properly (keep trying repeatedly for a few seconds while server boots up):
+```
+curl http://127.0.0.1:8080/ping
+#OR
+curl http://127.0.0.1:8081/models/
+```
+
+Install git-lfs to be able to download 🤗 models from the hub ([reference](https://stackoverflow.com/questions/71448559/git-large-file-storage-how-to-install-git-lfs-on-aws-ec2-linux-2-no-package)):
+```
+sudo yum install -y amazon-linux-extras
```

Comment:
Not sure how I feel about having instructions specific to Amazon Linux, Ubuntu is a much more common distro

Reply:
Sure, as decided in last comment, we shall cut out the `running on AWS EC2 instance` part from the `Image_classification_docker` example and keep it centered on Ubuntu distro.

I supposed a simple `pip install git-lfs` would do instead of these lines? ([link](https://pypi.org/project/git-lfs/))